### PR TITLE
JSON encode on  application/json

### PR DESCRIPTION
**Pull request by gotardo**
Autodetect the use of JSON content to change the post vars encoding to JSON in order to have a proper encoding on JSON communication.

### CODE REVIEW

File: src/Curl/Curl.php [hunk 1]
```diff
@@ -128,7 +128,10 @@ public function post($url, $data = array())
         $this->setOpt(CURLOPT_URL, $this->url);
         $this->setOpt(CURLOPT_CUSTOMREQUEST, 'POST');
         $this->setOpt(CURLOPT_POST, true);
-        $this->setOpt(CURLOPT_POSTFIELDS, $this->postfields($data));
+        if ($this->getHeader('Content-Type') === 'Content-Type: application/json')
```
Comment (sergeyklay) [inline src/Curl/Curl.php]: Why not `$this->json_pattern`?

**Comment by zborboa-google** [comment]
Thanks for the pull request. I believe what you are trying to accomplish has been implemented.
https://github.com/php-curl-class/php-curl-class/blob/master/src/Curl/Curl.php#L440

**Comment by gotardo** [comment]
Hi,
I see the problem: I am using composer to update my project dependencies. The last version in composer is 2.1.0, which still has not the fix. Are you planning to deliver this so we can take advantage of this update via composer?
Thanks!

**Comment by gotardo** [comment]
I also see that the current solution in v2.1.1 is failing to encode in JSON when passing a multidimensional array, for example, an array as a post var.

**Comment by zborboa-google** [comment]
@gotardo - [2.1.2](https://github.com/php-curl-class/php-curl-class/releases/tag/2.1.2) released which includes json request encoding (https://github.com/php-curl-class/php-curl-class/commit/80adf0f3a1c9f0ef9a5c2763675f7d3a4cb84c35).

**Comment by zborboa-google** [comment]
@gotardo - would this be an example of what you are describing?

```php
$curl = new Curl();
$curl->setHeader('Content-Type', 'application/json');
$curl->post(array(
    'foo' => array(
        'key' => 'val',
    ),
));
```

**Comment by gotardo** [comment]
Hi, yes, that's a similar case. One of the vars I need to post is a list of strings.

**Comment by zachborboa** [comment]
@gotardo - so like this?

```php
$curl->post(array(
    'key' => 'val',
    'strings' => array(
        'a',
        'b',
        'c',
    ),
));
```

**Comment by gotardo** [comment]
Yes,  like this

--
Gotardo González
PHP Consultant
contacto@gotardo.es
(+34) 616238080
GTalk/iChat: gotardogoque@gmail.com
Skype: gotardo.gonzalez

**Comment by zborboa-google** [comment]
@gotardo - can you check that the latest head fixes json-encoding of multidimensional arrays? Thank you

**Comment by zachborboa** [comment]
Should be fixed in 891a66c3275a98d8b26264019747a7fd67d368f8.

**Pull request closed without merging**